File: JavaFX/javafx-scala/src/main/scala/hello/Controller.scala
package hello

import scalafx.scene.control.TextArea
import scalafx.scene.text.{Text, Font}

class Controller (var done : Text , var scala : Text , var fx : Text , var use : Text , var ex : Text , var yours : TextArea ){

  private val fonts = "../fonts/bb.otf"::"../fonts/fo.otf"::"../fonts/comic.ttf"::"../fonts/hobbit.ttf"::"../fonts/nasa.ttf"::"../fonts/sc.ttf"::"../fonts/star.ttf"::Nil
  private var current_fonts = fonts

  def changeFx() : Unit = {
    val str = current_fonts.head
    val bf = Font.loadFont(getClass.getResource(str).toExternalForm, 50)
    val sf = Font.loadFont(getClass.getResource(str).toExternalForm, 30)
    done.setFont(bf)
    scala.setFont(bf)
    fx.setFont(bf)
    ex.setFont(sf)
    yours.setFont(sf)
    yours.requestFocus()

    if (current_fonts.tail == Nil){
      current_fonts = fonts
    } else {
      current_fonts = current_fonts.tail
    }
  }
}

File: JavaFX/javafx-java/src/sample/Controller.java
package sample;

import javafx.fxml.FXML;
import javafx.scene.control.TextArea;
import javafx.scene.text.Font;
import javafx.scene.text.Text;

import java.util.HashSet;
import java.util.Iterator;
import java.util.Set;

public class Controller {

    private Text done_txt;
    private Text java_txt;
    private Text fx_txt;
    private Text example;
    private TextArea yourText;

    private Set<String> fonts = new HashSet<>();

    private Iterator<String> it = fonts.iterator();

    public Controller(Text done, Text java, Text fx, Text ex, TextArea yours) {
        done_txt=done;
        java_txt=java;
        fx_txt=fx;
        fx_txt.setOnMouseClicked((e) -> changeFx());
        example=ex;
        yourText=yours;

        fonts.add("../fonts/bb.otf");
        fonts.add("../fonts/fo.otf");
        fonts.add("../fonts/comic.ttf");
        fonts.add("../fonts/hobbit.ttf");
        fonts.add("../fonts/nasa.ttf");
        fonts.add("../fonts/sc.ttf");
        fonts.add("../fonts/star.ttf");
        yourText.requestFocus();
    }

    public void changeFx() {
        if (!it.hasNext()) {
            it = fonts.iterator();
        }
        String font = it.next();
        Font bf = Font.loadFont(getClass().getResource(font).toExternalForm(), 50),
                sf = Font.loadFont(getClass().getResource(font).toExternalForm(), 30);
        done_txt.setFont(bf);
        java_txt.setFont(bf);
        fx_txt.setFont(bf);

        example.setFont(sf);
        yourText.setFont(sf);
        yourText.requestFocus();
    }
}